llvm/lib/Analysis/MemoryDependenceAnalysis.cpp

```diff
@@ -373,7 +373,10 @@ static bool canSkipClobberingStore(const StoreInst *SI,
     return false;
   if (MemoryLocation::get(SI).Size != MemLoc.Size)
     return false;
-  if (std::min(MemLocAlign, SI->getAlign()).value() < MemLoc.Size.getValue())
+  if (MemoryLocation::get(SI).Size.isScalable() || MemLoc.Size.isScalable())
```

Comment:
```suggestion
  if (MemLoc.Size.isScalable())
```
Sufficient, because both sizes are the same (checked in the previous line).